yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -290,8 +290,18 @@ private[yarn] class YarnAllocator(
       location: String,
       containersToUse: ArrayBuffer[Container],
       remaining: ArrayBuffer[Container]): Unit = {
+    // SPARK-6050: certain Yarn configurations return a resource configuration that doesn't match
+    // the request; for example, capacity scheduler + DefaultResourceCalculator. Allow users in
+    // those situations to disable resource matching.
+    val matchingResource =
+        if (sparkConf.getBoolean("spark.yarn.container.matchAnyResource", false)) {
```

Comment:
Sorry I'm still not seeing it. resource is:

Resource.newInstance(executorMemory + memoryOverhead, executorCores)

which is going to be executorCores passed in, which would be for instance 8 if I request 8. 

That resource is then passed to amClient.getMatchingRequests which is going to find requests that have 8 vcores, which isn't what we want because the RM without cpu scheduling returns ones with 1.